app/services/curation_sync_service.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class CurationSyncService
+  def self.sync(work_id)
```

Comment:
CodeClimate has a point here (that's not always a given). `CurationSyncService.sync` is a big method. I think we could start refactoring some of its logic into private methods. In order to do that, you'll probably need to refactor this into a proper class with state and make `sync` an instance method. So, for example, a `CurationSyncService` object could be initialized with a work to be synced, and then we'd call `CurationSyncService#sync` on the instance instead of the class (passing no argument).

Then, for example, we could extract the local variable assignment for `current_version_for_curation` into a private instance method. That would begin to simplify `CurationSyncService#sync` and reduce its responsibilities. We could probably do the same thing for some of the other local assignments.

It's reasonable for this class to be responsible for knowing what information it needs from a `Work` to do its job, but if we didn't want it to be responsible for also doing ActiveRecord stuff (pulling the work data out of the database and instantiating a `Work`), we could pass a `Work` object to the service's initializer (instead of an ID) and leave the data-finding responsibility to some other part of the code. That would reduce the overall responsibilities of this class and decouple it from ActiveRecord, making it a bit easier to test.

I'm not saying that it'll necessarily be the case here, but if a class ends up with enough private methods after a refactoring like this, and if it's not obvious that most of those private methods actually belong in some other existing part of the code, then it may become obvious that another whole class can be extracted from what was originally a single, large method.